File: ios/ios/Config/WeexSDKManager.m
//
//  WeexSDKManager.m
//  ios
//
//  Created by Michael on 2017/9/30.
//  Copyright © 2017年 weexstudy. All rights reserved.
//

#import "WeexSDKManager.h"
#import <WeexSDK/WeexSDK.h>
#import "WXImgLoaderDefaultImpl.h"
#import "WXEventModule.h"
#import "WXDeviceModule.h"
#import "WXHTTPModule.h"
#import "WXNaviBarModule.h"
#import "WXLOTAnimationView.h"
#import "WXPieChartView.h"
#import "WXWeChatModule.h"
#import "WXImagePickerModule.h"
#import "WXLineChartView.h"
#import "WXUserInfoModule.h"
#import "WXMyWebView.h"

@implementation WeexSDKManager

+(void)setup {
    [WXAppConfiguration setAppGroup:@"App"];
    [WXAppConfiguration setAppName:@"WeexDemo"];
    [WXAppConfiguration setAppVersion:@"1.0.0"];
    
    [WXSDKEngine initSDKEnvironment];
    
    // handler
    [WXSDKEngine registerHandler:[WXImgLoaderDefaultImpl new] withProtocol:@protocol(WXImgLoaderProtocol)];
    
    // module
    [WXSDKEngine registerModule:@"event" withClass:[WXEventModule class]];
    [WXSDKEngine registerModule:@"device" withClass:[WXDeviceModule class]];
    [WXSDKEngine registerModule:@"http" withClass:[WXHTTPModule class]];
    [WXSDKEngine registerModule:@"navbar" withClass:[WXNaviBarModule class]];
    [WXSDKEngine registerModule:@"wechat" withClass:[WXWeChatModule class]];
    [WXSDKEngine registerModule:@"imagePicker" withClass:[WXImagePickerModule class]];
    [WXSDKEngine registerModule:@"userInfo" withClass:[WXUserInfoModule class]];
    
    // component
    [WXSDKEngine registerComponent:@"animal" withClass:[WXLOTAnimationView class]];
    [WXSDKEngine registerComponent:@"pieChart" withClass:[WXPieChartView class]];
    [WXSDKEngine registerComponent:@"lineChart" withClass:[WXLineChartView class]];
    [WXSDKEngine registerComponent:@"web" withClass:[WXMyWebView class]];

#ifdef DEBUG
    [WXLog setLogLevel:WXLogLevelLog];
#endif
}

@end

File: Android/app/src/main/java/com/softbi/config/WeexSDKManager.java
package com.softbi.config;

import android.app.Application;
import android.content.Context;

import com.softbi.extend.adapter.ImageAdapter;
import com.softbi.extend.component.WXLOTAnimationView;
import com.softbi.extend.component.WXPieChartView;
import com.softbi.extend.module.WXDeviceModule;
import com.softbi.extend.module.WXEventModule;
import com.softbi.extend.module.WXHttpModule;
import com.softbi.extend.module.WXImagePickerModule;
import com.softbi.extend.module.WXNaviBarModule;
import com.softbi.extend.module.WXUserInfoModule;
import com.softbi.extend.module.WXWeChatModule;
import com.taobao.weex.InitConfig;
import com.taobao.weex.WXSDKEngine;

/**
 * Created by michael on 2018/3/15.
 */

public class WeexSDKManager {
    public static void setup(Application application) {
        InitConfig initConfig = new InitConfig.Builder().setImgAdapter(new ImageAdapter()).build();
        try {

            // weex 注册
            WXSDKEngine.initialize(application, initConfig);
            WXSDKEngine.registerModule("event", WXEventModule.class);
            WXSDKEngine.registerModule("device", WXDeviceModule.class);
            WXSDKEngine.registerModule("http", WXHttpModule.class);
            WXSDKEngine.registerModule("navbar", WXNaviBarModule.class);
            WXSDKEngine.registerModule("wechat", WXWeChatModule.class);
            WXSDKEngine.registerModule("imagePicker", WXImagePickerModule.class);
            WXSDKEngine.registerModule("userInfo", WXUserInfoModule.class);

            WXSDKEngine.registerComponent("pieChart", WXPieChartView.class);
            WXSDKEngine.registerComponent("animal", WXLOTAnimationView.class);

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
